dashboard/client/src/pages/log/Logs.tsx

```diff
@@ -25,126 +34,235 @@ const useStyles = makeStyles((theme) => ({
+                defaultValue={fileName}
+                label="File Name"
+                onChange={(val) => {
+                  setFileName(val);
+                }}
+              />
+            </div>
+          )}
+        </Paper>
+        <Paper>
+          {nodeId ? (
+            <StateApiLogsFilesList
+              nodeId={nodeId}
+              folder={folder}
+              fileName={fileName}
+            />
+          ) : (
+            <StateApiLogsNodesList />
+          )}
+        </Paper>
+      </TitleCard>
+    </div>
+  );
 };
 
-const Logs = () => {
+export const StateApiLogsNodesList = () => {
+  const { data: nodes, error } = useSWR(["/api/v0/nodes"], async () => {
+    const resp = await getNodeList();
+    return resp.data.data.summary;
+  });
+
+  const isLoading = nodes === undefined && error === undefined;
+
+  return (
+    <React.Fragment>
+      {isLoading ? (
+        <Typography>Loading...</Typography>
+      ) : error ? (
+        <Typography color="error">{error}</Typography>
+      ) : (
+        <List>
+          {nodes?.map(({ raylet: { nodeId }, ip }) => (
+            <ListItem key={nodeId}>
+              <Link to={`?nodeId=${nodeId}`}>
+                Node ID: {nodeId} (IP: {ip})
+              </Link>
+            </ListItem>
+          ))}
+        </List>
+      )}
+    </React.Fragment>
+  );
+};
+
+type StateApiLogsFilesListProps = {
+  nodeId: string;
+  folder: string | null;
+  fileName: string;
+};
+
+export const StateApiLogsFilesList = ({
+  nodeId,
+  folder,
+  fileName,
+}: StateApiLogsFilesListProps) => {
+  // We want to do a partial search for file name.
+  const fileNameGlob = fileName ? `*${fileName}*` : undefined;
+  const glob = fileNameGlob
+    ? folder
+      ? `${folder}/${fileNameGlob}`
+      : `${fileNameGlob}`
+    : folder
+    ? `${folder}/*`
+    : undefined;
+
+  const { data: fileGroups, error } = useSWR(
+    nodeId ? ["/api/v0/logs", nodeId, glob] : null,
+    async ([_, nodeId, glob]) => {
+      const resp = await listStateApiLogs({ nodeId, glob });
+      return resp.data.data.result;
+    },
+  );
+
+  const isLoading = fileGroups === undefined && error === undefined;
+  const files =
+    fileGroups !== undefined
+      ? Object.values(fileGroups)
+          .flatMap((e) => e)
+          .sort()
+      : [];
+
+  return (
+    <React.Fragment>
+      {isLoading ? (
+        <p>Loading...</p>
+      ) : (
+        files.length === 0 && <p>No files found.</p>
+      )}
+      {files && (
+        <List>
+          {files.map((fileName) => {
+            const isDir = fileName.endsWith("/");
+            const fileNameWithoutEndingSlash = fileName.substring(
+              0,
+              fileName.length - 1,
+            );
+            const parentFolder = folder ? `${folder}/` : "";
+            const fileNameWithoutParent = fileName.startsWith(parentFolder)
+              ? fileName.substring(parentFolder.length)
+              : fileName;
```

Comment:
To confirm, the files listed by `listStateApiLogs` could be either filepath or filename, so that's why we need this branching, right?